File: examples/union_find_tree.yosupo.test.cpp
#define PROBLEM "https://judge.yosupo.jp/problem/unionfind"
#include <iostream>
#include "./union_find_tree.hpp"
#include "./macros.hpp"
using namespace std;

int main() {
    int n, q; cin >> n >> q;
    union_find_tree uft(n);
    REP (i, q) {
        int t, u, v; cin >> t >> u >> v;
        if (t == 0) {
            uft.unite_trees(u, v);
        } else if (t == 1) {
            cout << uft.is_same(u, v) << endl;
        }
    }
    return 0;
}

File: examples/nim/union_find_tree_yosupo_test.nim
# verification-helper: PROBLEM https://judge.yosupo.jp/problem/unionfind

include "examples/nim/headers.nim"
from examples/nim/union_find_tree import unite_trees, initUnionFindTree, is_same

let 
  n,q = nextInt()
var uft = initUnionFindTree(n)

for i in 0..<q:
  let t, u, v = nextInt()
  if t == 0:
    uft.unite_trees(u, v)
  elif t == 1:
    echo if uft.is_same(u, v): 1 else: 0
